tools/readlink.py

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+"""Canonicalizes a path and follows any symlinks.
+
+This is the equivalent of `readlink -f` on many Linux systems. This is
+useful as there are often differences in readlink on different
+platforms.
+
+"""
+from __future__ import print_function
+import os
+import sys
+
+print(os.path.realpath(sys.argv[1]))
```

Comment:
I don't think it calls `readlink -f` internally. Tests pass now and before they didn't.

Additionally, there are no restrictions on availability listed on [python.org](https://docs.python.org/2/library/os.path.html#os.path.realpath). At the very least, Python isn't documenting any differences in behavior for this function and (when compared to the rest of the module) is implicitly claiming it works on Windows where the shell command `readlink` isn't available.